tpl/tplimpl/template_funcs_test.go

```diff
@@ -1021,6 +1023,52 @@ func TestIntersect(t *testing.T) {
 	}
 }
 
+func TestUnion(t *testing.T) {
+	t.Parallel()
+	for i, this := range []struct {
+		sequence1 interface{}
+		sequence2 interface{}
+		expect    interface{}
+	}{
+		{[]string{"a", "b", "c", "c"}, []string{"a", "b", "b"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b"}, []string{"a", "b", "c"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b", "c"}, []string{"d", "e"}, []string{"a", "b", "c", "d", "e"}},
+		{[]string{}, []string{}, []string{}},
+		{[]string{"a", "b"}, nil, make([]interface{}, 0)},
+		{nil, []string{"a", "b"}, make([]interface{}, 0)},
+		{nil, nil, make([]interface{}, 0)},
+		{[]string{"1", "2"}, []int{1, 2}, make([]string, 0)},
```

Comment:
I also think this is a surprising result. One could argue that the result should be either

* "1", "2", 1, 2 (which is impossible)
* "1", "2"
* 1, 2
* error

But not an empty string slice. Probably easiest to make it an error, not sure.